File: src/main/java/no/nav/sbl/dialogarena/modiasyforest/rest/domain/sykepengesoknad/Sykepengesoknad.java
package no.nav.sbl.dialogarena.modiasyforest.rest.domain.sykepengesoknad;

import no.nav.sbl.dialogarena.modiasyforest.rest.domain.Arbeidsgiver;
import no.nav.sbl.dialogarena.modiasyforest.rest.domain.sykepengesoknad.oppsummering.Oppsummering;

import java.time.LocalDate;
import java.util.List;

public class Sykepengesoknad {

    public String id;
    public String sykmeldingId;
    public String status;
    public LocalDate opprettetDato;
    public LocalDate avbruttDato;
    public Arbeidsgiver arbeidsgiver;
    public LocalDate identdato;
    public boolean ansvarBekreftet;
    public boolean bekreftetKorrektInformasjon;
    public String arbeidsgiverForskutterer;
    public List<Datospenn> egenmeldingsperioder;
    public LocalDate gjenopptattArbeidFulltUtDato;
    public List<Datospenn> ferie;
    public List<Datospenn> permisjon;
    public Utenlandsopphold utenlandsopphold;
    public List<Aktivitet> aktiviteter;
    public List<AnnenInntektskilde> andreInntektskilder;
    public Utdanning utdanning;
    public LocalDate sykmeldingSkrevetDato;
    public LocalDate sendtTilArbeidsgiverDato;
    public LocalDate sendtTilNAVDato;
    public LocalDate forrigeSykeforloepTom;
    public LocalDate forrigeSendteSoknadTom;
    public String korrigerer;
    public LocalDate fom;
    public LocalDate tom;
    public int del;
    public Oppsummering oppsummering;

    public Sykepengesoknad withOppsummering(final Oppsummering oppsummering) {
        this.oppsummering = oppsummering;
        return this;
    }

    public Sykepengesoknad withAnsvarBekreftet(final boolean ansvarBekreftet) {
        this.ansvarBekreftet = ansvarBekreftet;
        return this;
    }

    public Sykepengesoknad withEgenmeldingsperioder(final List<Datospenn> egenmeldingsperioder) {
        this.egenmeldingsperioder = egenmeldingsperioder;
        return this;
    }

    public Sykepengesoknad withGjenopptattArbeidFulltUtDato(final LocalDate gjenopptattArbeidFulltUtDato) {
        this.gjenopptattArbeidFulltUtDato = gjenopptattArbeidFulltUtDato;
        return this;
    }

    public Sykepengesoknad withFerie(final List<Datospenn> ferie) {
        this.ferie = ferie;
        return this;
    }

    public Sykepengesoknad withPermisjon(final List<Datospenn> permisjon) {
        this.permisjon = permisjon;
        return this;
    }

    public Sykepengesoknad withUtenlandsopphold(final Utenlandsopphold utenlandsopphold) {
        this.utenlandsopphold = utenlandsopphold;
        return this;
    }

    public Sykepengesoknad withUtdanning(final Utdanning utdanning) {
        this.utdanning = utdanning;
        return this;
    }

    public Sykepengesoknad withId(final String id) {
        this.id = id;
        return this;
    }

    public Sykepengesoknad withStatus(String status) {
        this.status = status;
        return this;
    }

    public Sykepengesoknad withOpprettetDato(LocalDate opprettet) {
        this.opprettetDato = opprettet;
        return this;
    }

    public Sykepengesoknad withAndreInntektskilder(final List<AnnenInntektskilde> andreInntektskilder) {
        this.andreInntektskilder = andreInntektskilder;
        return this;
    }

    public Sykepengesoknad withBekreftetKorrektInformasjon(final boolean bekreftetKorrektInformasjon) {
        this.bekreftetKorrektInformasjon = bekreftetKorrektInformasjon;
        return this;
    }

    public Sykepengesoknad withIdentdato(final LocalDate identdato) {
        this.identdato = identdato;
        return this;
    }

    public Sykepengesoknad withAvbruttDato(LocalDate avbruttDato) {
        this.avbruttDato = avbruttDato;
        return this;
    }

    public Sykepengesoknad withAktiviteter(final List<Aktivitet> aktiviteter) {
        this.aktiviteter = aktiviteter;
        return this;
    }

    public Sykepengesoknad withArbeidsgiver(final Arbeidsgiver arbeidsgiver) {
        this.arbeidsgiver = arbeidsgiver;
        return this;
    }

    public Sykepengesoknad withArbeidsgiverForskutterer(final String arbeidsgiverForskutterer) {
        this.arbeidsgiverForskutterer = arbeidsgiverForskutterer;
        return this;
    }

    public Sykepengesoknad withSykmeldingSkrevetDato(LocalDate sykmeldingSkrevetDato) {
        this.sykmeldingSkrevetDato = sykmeldingSkrevetDato;
        return this;
    }

    public Sykepengesoknad withSendtTilArbeidsgiverDato(final LocalDate sendtTilArbeidsgiverDato) {
        this.sendtTilArbeidsgiverDato = sendtTilArbeidsgiverDato;
        return this;
    }

    public Sykepengesoknad withSendtTilNAVDato(final LocalDate sendtTilNAVDato) {
        this.sendtTilNAVDato = sendtTilNAVDato;
        return this;
    }

    public Sykepengesoknad withForrigeSykeforloepTom(final LocalDate forrigeSykeforloepTom) {
        this.forrigeSykeforloepTom = forrigeSykeforloepTom;
        return this;
    }

    public Sykepengesoknad withKorrigerer(final String korrigerer) {
        this.korrigerer = korrigerer;
        return this;
    }

    public Sykepengesoknad withFom(final LocalDate fom) {
        this.fom = fom;
        return this;
    }

    public Sykepengesoknad withTom(final LocalDate tom) {
        this.tom = tom;
        return this;
    }

    public Sykepengesoknad withDel(final int del) {
        this.del = del;
        return this;
    }

    public Sykepengesoknad withForrigeSendteSoknadTom(final LocalDate forrigeSendteSoknadTom) {
        this.forrigeSendteSoknadTom = forrigeSendteSoknadTom;
        return this;
    }

    public Sykepengesoknad withSykmeldingId(final String sykmeldingId) {
        this.sykmeldingId = sykmeldingId;
        return this;
    }
}

File: src/main/kotlin/no/nav/syfo/controller/domain/sykepengesoknad/Sykepengesoknad.kt
package no.nav.syfo.controller.domain.sykepengesoknad

import no.nav.syfo.controller.domain.Arbeidsgiver
import no.nav.syfo.controller.domain.sykepengesoknad.oppsumering.Oppsummering
import java.io.Serializable
import java.time.LocalDate

class Sykepengesoknad : Serializable {
    var id: String? = null
    var sykmeldingId: String? = null
    var status: String? = null
    var opprettetDato: LocalDate? = null
    var avbruttDato: LocalDate? = null
    var arbeidsgiver: Arbeidsgiver? = null
    var identdato: LocalDate? = null
    var ansvarBekreftet = false
    var bekreftetKorrektInformasjon = false
    var arbeidsgiverForskutterer: String? = null
    var egenmeldingsperioder: List<Datospenn>? = null
    var gjenopptattArbeidFulltUtDato: LocalDate? = null
    var ferie: List<Datospenn>? = null
    var permisjon: List<Datospenn>? = null
    var utenlandsopphold: Utenlandsopphold? = null
    var aktiviteter: List<Aktivitet>? = null
    var andreInntektskilder: List<AnnenInntektskilde>? = null
    var utdanning: Utdanning? = null
    var sykmeldingSkrevetDato: LocalDate? = null
    var sendtTilArbeidsgiverDato: LocalDate? = null
    var sendtTilNAVDato: LocalDate? = null
    var forrigeSykeforloepTom: LocalDate? = null
    var forrigeSendteSoknadTom: LocalDate? = null
    var korrigerer: String? = null
    var fom: LocalDate? = null
    var tom: LocalDate? = null
    var del = 0
    var oppsummering: Oppsummering? = null
    fun withOppsummering(oppsummering: Oppsummering?): Sykepengesoknad {
        this.oppsummering = oppsummering
        return this
    }

    fun withAnsvarBekreftet(ansvarBekreftet: Boolean): Sykepengesoknad {
        this.ansvarBekreftet = ansvarBekreftet
        return this
    }

    fun withEgenmeldingsperioder(egenmeldingsperioder: List<Datospenn>?): Sykepengesoknad {
        this.egenmeldingsperioder = egenmeldingsperioder
        return this
    }

    fun withGjenopptattArbeidFulltUtDato(gjenopptattArbeidFulltUtDato: LocalDate?): Sykepengesoknad {
        this.gjenopptattArbeidFulltUtDato = gjenopptattArbeidFulltUtDato
        return this
    }

    fun withFerie(ferie: List<Datospenn>?): Sykepengesoknad {
        this.ferie = ferie
        return this
    }

    fun withPermisjon(permisjon: List<Datospenn>?): Sykepengesoknad {
        this.permisjon = permisjon
        return this
    }

    fun withUtenlandsopphold(utenlandsopphold: Utenlandsopphold?): Sykepengesoknad {
        this.utenlandsopphold = utenlandsopphold
        return this
    }

    fun withUtdanning(utdanning: Utdanning?): Sykepengesoknad {
        this.utdanning = utdanning
        return this
    }

    fun withId(id: String?): Sykepengesoknad {
        this.id = id
        return this
    }

    fun withStatus(status: String?): Sykepengesoknad {
        this.status = status
        return this
    }

    fun withOpprettetDato(opprettet: LocalDate?): Sykepengesoknad {
        opprettetDato = opprettet
        return this
    }

    fun withAndreInntektskilder(andreInntektskilder: List<AnnenInntektskilde>?): Sykepengesoknad {
        this.andreInntektskilder = andreInntektskilder
        return this
    }

    fun withBekreftetKorrektInformasjon(bekreftetKorrektInformasjon: Boolean): Sykepengesoknad {
        this.bekreftetKorrektInformasjon = bekreftetKorrektInformasjon
        return this
    }

    fun withIdentdato(identdato: LocalDate?): Sykepengesoknad {
        this.identdato = identdato
        return this
    }

    fun withAvbruttDato(avbruttDato: LocalDate?): Sykepengesoknad {
        this.avbruttDato = avbruttDato
        return this
    }

    fun withAktiviteter(aktiviteter: List<Aktivitet>?): Sykepengesoknad {
        this.aktiviteter = aktiviteter
        return this
    }

    fun withArbeidsgiver(arbeidsgiver: Arbeidsgiver?): Sykepengesoknad {
        this.arbeidsgiver = arbeidsgiver
        return this
    }

    fun withArbeidsgiverForskutterer(arbeidsgiverForskutterer: String?): Sykepengesoknad {
        this.arbeidsgiverForskutterer = arbeidsgiverForskutterer
        return this
    }

    fun withSykmeldingSkrevetDato(sykmeldingSkrevetDato: LocalDate?): Sykepengesoknad {
        this.sykmeldingSkrevetDato = sykmeldingSkrevetDato
        return this
    }

    fun withSendtTilArbeidsgiverDato(sendtTilArbeidsgiverDato: LocalDate?): Sykepengesoknad {
        this.sendtTilArbeidsgiverDato = sendtTilArbeidsgiverDato
        return this
    }

    fun withSendtTilNAVDato(sendtTilNAVDato: LocalDate?): Sykepengesoknad {
        this.sendtTilNAVDato = sendtTilNAVDato
        return this
    }

    fun withForrigeSykeforloepTom(forrigeSykeforloepTom: LocalDate?): Sykepengesoknad {
        this.forrigeSykeforloepTom = forrigeSykeforloepTom
        return this
    }

    fun withKorrigerer(korrigerer: String?): Sykepengesoknad {
        this.korrigerer = korrigerer
        return this
    }

    fun withFom(fom: LocalDate?): Sykepengesoknad {
        this.fom = fom
        return this
    }

    fun withTom(tom: LocalDate?): Sykepengesoknad {
        this.tom = tom
        return this
    }

    fun withDel(del: Int): Sykepengesoknad {
        this.del = del
        return this
    }

    fun withForrigeSendteSoknadTom(forrigeSendteSoknadTom: LocalDate?): Sykepengesoknad {
        this.forrigeSendteSoknadTom = forrigeSendteSoknadTom
        return this
    }

    fun withSykmeldingId(sykmeldingId: String?): Sykepengesoknad {
        this.sykmeldingId = sykmeldingId
        return this
    }
}